toolkit/tools/pkg/imagecustomizerlib/customizeusers.go

```diff
@@ -72,12 +75,16 @@ func addOrUpdateUser(user imagecustomizerapi.User, baseConfigPath string, imageC
 			password = string(passwordFileContents)
 		}
 
-		hashedPassword = password
-		if !passwordIsHashed {
-			// Hash the password.
-			hashedPassword, err = userutils.HashPassword(password)
-			if err != nil {
-				return err
+		// Only proceed if password is not empty
+		if strings.TrimSpace(password) != "" {
```

Comment:
If the user specifies the password field but provides an empty password value, then we should remove the password from the user. If the user doesn't provide the password field, then the user's password should be unchanged.